xc7/make/project_xray.cmake

```diff
@@ -377,53 +213,233 @@ function(PROJECT_XRAY_PREPARE_DATABASE)
   get_target_property_required(PYTHON3 env PYTHON3)
   get_target_property(PYTHON3_TARGET env PYTHON3_TARGET)
 
-  set(PART ${PROJECT_XRAY_PREPARE_DATABASE_PART})
+  set(ARCH ${PROJECT_XRAY_PREPARE_DATABASE_ARCH})
+  set(PROTOTYPE_PART ${PROJECT_XRAY_PREPARE_DATABASE_PROTOTYPE_PART})
+
   set(FORM_CHANNELS ${symbiflow-arch-defs_SOURCE_DIR}/xc7/utils/prjxray_form_channels.py)
   set(ASSIGN_PINS ${symbiflow-arch-defs_SOURCE_DIR}/xc7/utils/prjxray_assign_tile_pin_direction.py)
-  file(GLOB DEPS ${PRJXRAY_DB_DIR}/${PART}/*.json)
-  file(GLOB DEPS2 ${PRJXRAY_DIR}/prjxray/*.py)
-
-  set(CHANNELS channels.db)
-  add_custom_command(
-    OUTPUT ${CHANNELS}
-    COMMAND ${CMAKE_COMMAND} -E env PYTHONPATH=${PRJXRAY_DIR}:${symbiflow-arch-defs_SOURCE_DIR}/utils
-    ${PYTHON3} ${FORM_CHANNELS}
-    --db_root ${PRJXRAY_DB_DIR}/${PART}/
-    --connection_database ${CMAKE_CURRENT_BINARY_DIR}/${CHANNELS}
-    DEPENDS
-    ${FORM_CHANNELS}
-    ${DEPS} ${DEPS2} simplejson progressbar2 intervaltree hilbertcurve
-    ${PYTHON3} ${PYTHON3_TARGET}
-    )
+  file(GLOB DEPS ${PRJXRAY_DB_DIR}/${ARCH}/*.json)
+  file(GLOB DEPS2 ${PRJXRAY_DB_DIR}/${ARCH}/${PART}/*.json)
+  file(GLOB DEPS3 ${PRJXRAY_DIR}/prjxray/*.py)
+
+  file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/channels)
+  foreach(PART ${PROJECT_XRAY_PREPARE_DATABASE_PARTS})
+    file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/channels/${PART})
+    set(CHANNELS channels/${PART}/channels.db)
+    add_custom_command(
+      OUTPUT ${CHANNELS}
+      COMMAND ${CMAKE_COMMAND} -E env PYTHONPATH=${PRJXRAY_DIR}:${symbiflow-arch-defs_SOURCE_DIR}/utils
+      ${PYTHON3} ${FORM_CHANNELS}
+      --db_root ${PRJXRAY_DB_DIR}/${ARCH}/
+      --part ${PROTOTYPE_PART}
+      --connection_database ${CMAKE_CURRENT_BINARY_DIR}/${CHANNELS}
```

Comment:
Why are we generating a database per part?

Reply:
So, `PARTS` variable contains a list of parts sub-families (50T and 200T). As of my understanding, each sub-family needs its own channels.

Reply:
Ok.